python/mp/rosary.py

```diff
@@ -57,9 +87,29 @@ def __init__(self, ip="127.0.0.1", port=5005, dispatcher=None, name="rosary"):
         self.dispatcher = dispatcher
         # Available knobs to turn
         self.knobs = {}
+        self.updater_list = []
 
+        self.osc_client = udp_client.UDPClient(self.osc_ip, self.osc_port)
+
+        # create the three classes of LED "beads"
         for i in range(self.BEAD_COUNT):
             self.beads.append(Bead(i))
+        self.updater_list.append(Updater(name='rosary',
+                                         bead_list=self.beads,
+                                         osc_client=self.osc_client))
+
+        for i in range(self.BASE_COUNT):
+            self.bases.append(Bead(i))
+        self.updater_list.append(Updater(name='base',
+                                         bead_list=self.bases,
+                                         osc_client=self.osc_client))
+
+        for i in range(self.CROSS_LED_COUNT):
+            self.cross.append(Bead(i))
+        self.updater_list.append(Updater(name='cross',
+                                         bead_list=self.cross,
+                                         osc_client=self.osc_client))
+
```

Comment:
Shouldn't the bead class be taking a name arg too now? Unless I'm missing something, just reading this it seems self.bases[0], self.beads[0], and self.cross[0] would all refer to the same bead?